### PR TITLE
tx api: add address, type, value to vin. add fee.

### PR DESCRIPTION
**Pull request by pointbiz**
This pull requests adds the address, value and type information to vin for a transaction api results. This way the caller does not have to make multiple calls to get the vin information to get addresses and calculate the fee. I have calculated the fee and added it to the results as well.

This scenario came up when attempting to consume the tx API in my bitcoin-reports.com project.

### CODE REVIEW

**Comment by dykstranet** [comment]
I tested this and, indeed it gives trx value and fee value that I needed. Would be nice to see this merged. Thanks for the help!

File: routes/apiRouter.js [hunk 1]
```diff
@@ -126,10 +126,34 @@ router.get("/tx/:txid", function(req, res, next) {
...
 	var promises = [];
 
-	promises.push(coreApi.getRawTransactionsWithInputs([txid]));
+	let txInputLimit = (res.locals.crawlerBot) ? 3 : -1;
+
+	promises.push(coreApi.getRawTransactionsWithInputs([txid], txInputLimit));
 
 	Promise.all(promises).then(function(results) {
-		res.json(results[0].transactions[0]);
+		var outJson = results[0].transactions[0];
```
Comment (dykstranet) [inline routes/apiRouter.js:134]: We probably should use `const`/`let` construct instead `var` for modern JS, just like the rest of the code.

**Comment by pointbiz** [comment]
@janoside 
Would you have time to consider this PR for the next release?

File: routes/apiRouter.js (hunk 1 above)
Comment (jsarenik) [reply routes/apiRouter.js:134]: @pointbiz Please change your patch accordingly, `git commit --amend` it and force push. Not resolved yet.

Comment (pointbiz) [reply routes/apiRouter.js:134]: The codebase uses a mix of var and let. I don't follow why this one line needs changing.

Comment (pointbiz) [reply routes/apiRouter.js]: I made the requested changes regarding `var` vs `const`/`let`.

**Comment by herb562** [comment]
This stuff's too complicated man stop emailing me please

**Comment by herb562** [comment]
FUCK this stuff I don't understand please stop emailing me to

**Comment by herb562** [comment]
FUCK got hud

**Comment by herb562** [comment]
To fuck btc

**Pull request merged**